compiler/kernel/filter.go

```diff
@@ -23,20 +23,34 @@ func (f *Filter) AsEvaluator() (expr.Evaluator, error) {
 		return nil, nil
 	}
 	b := f.builder
-	return compileFilter(b.pctx.Zctx, b.scope, f.pushdown)
+	return compileFilter(b.pctx.Zctx, f.pushdown)
 }
 
 func (f *Filter) AsBufferFilter() (*expr.BufferFilter, error) {
 	if f == nil {
 		return nil, nil
 	}
-	return CompileBufferFilter(f.pushdown)
+	return CompileBufferFilter(f.builder.pctx.Zctx, f.pushdown)
 }
 
-func compileCompareField(zctx *zed.Context, scope *Scope, e *dag.BinaryExpr) (expr.Evaluator, error) {
+func isLiteral(zctx *zed.Context, e dag.Expr) (*zed.Value, error) {
```

Comment:
1. Nit: `compileCompareField` is more interesting so put this after it.
2. `isLiteral` doesn't feel like the right name for something that doesn't return a Boolean. How about `asLiteral`, `parseLiteral`, `maybeParseLiteral`, or `tryParseLiteral`?

Reply:
On 2), I use this naming pattern because the `val!=nil` condition is a boolean indicator.  You get back nil if and only if it *is* not the thing being looed for so I think the word "is" is appropriate.